### DISCUSSION TITLE
Macro expansion

### DISCUSSION THREAD

**Discussion by Almaz5200** [wojciech-kulik/xcodebuild.nvim#194] [Q&A]
I wonder if it's possible to somehow see macro expansion in neovim? Closest thing I was able to find is "Inline macro" in code actions. That works I guess, but would be cool to be able to actually just preview it

**Comment by wojciech-kulik** [comment]
I'm not really familiar with macros, but if it's possible to get from somewhere the expanded version, then the plugin could inject it using virtual text.

**Reply by Almaz5200** [reply]
I'm not familiar with how LSP's work, but I've found [this PR](https://github.com/swiftlang/sourcekit-lsp/pull/1479) that suggests there is a way, it also looks like there's that option in vscode swift plugin

**Reply by wojciech-kulik** [reply]
I checked and this function is not yet available. It will be in Swift 6.0.1. I mean, you can download the snapshot and change it in Xcode, but it will officially appear with the new Xcode, probably with Xcode 16.1.

**Reply by Almaz5200** [reply]
I see, thanks. Then for now I’ll just expand and undo when I need it  :D

**Reply by wojciech-kulik** [reply]
If you want to play with the new function you can try switching Swift: https://github.com/wojciech-kulik/xcodebuild.nvim/discussions/82

**Reply by Almaz5200** [reply]
I've tried to, but unfortunately none of my projects build with it yet :D

**Reply by PangMo5** [reply]
https://forums.swift.org/t/gsoc-2024-expansion-of-swift-macros-in-visual-studio-code-detailed-post-lokesh-t-r-alex-hoppen-adam-fowler/74163

Based on that article, it looks like can use the extension through CodeAction, and it seems to work well.
It has a slight UX inconvenience, but doesn't seem to be a big deal.

<img width="1824" alt="image" src="https://github.com/user-attachments/assets/7e8e6057-b46f-4b8d-aa29-78a24b673468" />
<img width="1824" alt="image" src="https://github.com/user-attachments/assets/db9ddcf9-cf54-4016-b456-32e9b978d5cb" />
<img width="1824" alt="image" src="https://github.com/user-attachments/assets/c0901050-5d70-4494-9163-cbdd152c5470" />
